indico/modules/rb/client/js/common/rooms/edit/RoomEditLocation.jsx

```diff
@@ -0,0 +1,114 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2020 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import {Form, Header, Tab} from 'semantic-ui-react';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {Translate} from 'indico/react/i18n';
+import {FinalInput, parsers as p, validators as v} from 'indico/react/forms';
+
+function RoomEditLocation({active}) {
+  return (
+    <Tab.Pane active={active}>
+      <Header>
+        <Translate>Location</Translate>
+      </Header>
+      <FinalInput
+        fluid
+        name="verbose_name"
+        label={Translate.string('Name')}
+        required={false}
+        nullIfEmpty
+        hideErrorPopup={!active}
+      />
+      <Form.Group widths="four">
+        <Form.Field width={8}>
+          <FinalInput
+            name="site"
+            label={Translate.string('Site')}
+            hideErrorPopup={!active}
+            required
+          />
+        </Form.Field>
+        <FinalInput
+          name="building"
+          label={Translate.string('Building')}
+          hideErrorPopup={!active}
+          required
+        />
+        <FinalInput
+          name="floor"
+          label={Translate.string('Floor')}
+          hideErrorPopup={!active}
+          required
+        />
+        <FinalInput
+          name="number"
+          label={Translate.string('Number')}
+          hideErrorPopup={!active}
+          required
+        />
+      </Form.Group>
+      <Form.Group widths="equal">
+        <FinalInput
+          fluid
+          type="number"
+          name="surface_area"
+          label={Translate.string('Surface Area (m²)')}
+          validate={v.optional(v.min(0))}
+          hideErrorPopup={!active}
+        />
+        <FinalInput
+          fluid
+          type="text"
+          name="latitude"
+          label={Translate.string('Latitude')}
+          parse={f => p.number(f, false)}
+          validate={v.optional(v.number())}
```

Comment:
Not sure if I'm just missing it, but while latitude/longitude are both optional, once one of them is set, the other becomes required as well.

Reply:
You might be able to wrap both of these components in a `<Field name="theotherfieldsname" subscription={{value: true}}>` to have access to the other field's value, and then set `validate` based on this.

Reply:
Unfortunately, even though this is the simplest solution, it will not work properly. It seems that the validation happens before the field can change it again.
One simple test case is:
- Set latitude to 123
- Set longitude to 123
- Remove latitude
- Remove longitude
- Latitude will remain required

I believe the alternative would be moving this to an after-submit validation. I'm open for suggestions.

Reply:
validating it on the server side is better than nothing for sure (and i hope we already do that?!).

or you could use a form-level validation function just for this case. i think this would actually be logic: validate field-specific stuff on the field-level, and stuff spanning multiple fields on the form level

Reply:
> validating it on the server side is better than nothing for sure (and i hope we already do that?!).
> 
> and you could use a form-level validation function just for this case. i think this would actually be logic: validate field-specific stuff on the field-level, and stuff spanning multiple fields on the form level

Agree, I'll make this the solution.